### PR TITLE
SLVS-761 Rules should not throw exceptions on invalid inputs

### PR DESCRIPTION
**Pull request by tvsonar**

### CODE REVIEW

File: src/SonarLint.CSharp/Rules/RedundantArgument.cs [hunk 1]
```diff
@@ -115,6 +115,19 @@ public ArgumentParameterMapping(ArgumentSyntax argument, IParameterSymbol parame
                 Argument = argument;
                 Parameter = parameter;
             }
+
+            internal static ArgumentParameterMapping GetParameterMapping(ArgumentSyntax argument, MethodParameterLookup methodParameterLookup)
```
Comment (tvsonar) [inline src/SonarLint.CSharp/Rules/RedundantArgument.cs]: As this method and the class is shared by many analyzers, we could put it inside the `MethodParameterLookup` class.

File: src/SonarLint.CSharp/Helpers/MethodParameterLookup.cs [hunk 2]
```diff
@@ -44,18 +45,22 @@ public IMethodSymbol MethodSymbol
             }
         }
 
-        public static IParameterSymbol GetParameterSymbol(ArgumentSyntax argument, ArgumentListSyntax argumentList, IMethodSymbol method)
+        public static bool TryGetParameterSymbol(ArgumentSyntax argument, ArgumentListSyntax argumentList,
+            IMethodSymbol method, out IParameterSymbol parameter)
         {
+            parameter = null;
             if (!argumentList.Arguments.Contains(argument) ||
```
Comment (dbolkensteyn) [inline src/SonarLint.CSharp/Helpers/MethodParameterLookup.cs]: @tvsonar minor but it seems that this should be the last check, as the ones on the method symbol are going to be super fast

File: src/SonarLint.CSharp/Rules/RedundantArgument.cs (hunk 1 above)
Comment (dbolkensteyn) [inline src/SonarLint.CSharp/Rules/RedundantArgument.cs]: then let's do so?

**Comment by dbolkensteyn** [comment]
@tvsonar I think the `GetParameterMapping()` method should be removed. Could you also update the Jira ticket to make it more user friendly? (i.e. document the patterns that are fixed by this ticket and PR)

**Comment by dbolkensteyn** [comment]
LGTM

**Pull request merged**